ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,1104 @@
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "#!import \"ReportScopes\""
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "using Systemorph.Vertex.Pivot.Builder.Interfaces;",
+                "\nusing Systemorph.InteractiveObjects;",
+                "\nusing Systemorph.Vertex.Session;",
+                "\nusing Systemorph.Vertex.Export;",
+                "\nusing Systemorph.Vertex.Export.Factory;",
+                "\nusing Systemorph.Vertex.InteractiveObjects;"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Form Entity Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Helper Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface MutableScopeWithWorkspace<TStorage> : IMutableScopeWithStorage<TStorage> ",
+                "\nwhere TStorage : ReportStorage",
+                "\n{",
+                "\n    protected IWorkspace workspace => GetStorage().Workspace;   ",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Currency Type"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface CurrencyFormsEntity : IMutableScope {",
+                "\n    [DropdownEnum(typeof(CurrencyType))]",
+                "\n    CurrencyType CurrencyType { get; set; } ",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Scenario"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface ScenarioFormsEntity<TStorage> : MutableScopeWithWorkspace<TStorage> ",
```

Comment:
same here, why is this scope generic if the type can only be assignable to ReportStorage? Do we intend to have different storage types inheriting from ReportStorage? Is this the case at the moment?

Reply:
Yes, this is exactly the case. In particular, in the U+ plus project we have a different ReportStorage, that is different from this one but inherits from it. The scope should be usable with both though. And other projects might have different needs and will have their own storages, that inherit from the generic one.